### PR TITLE
apis: move qos utils for extensions

### PR DESCRIPTION
**Pull request by saintube**
### Ⅰ. Describe what this PR does

1. Move QoS utility functions into another file for extensions. (e.g. customize the annotation of pod QoS)
2. Revise some comments in the API package.

### V. Checklist

- [X] I have written necessary docs and comments
- [X] I have added necessary unit tests and integration tests
- [X] All checks passed in `make test`

### CODE REVIEW

File: apis/extension/qos.go [hunk 1]
```diff
@@ -29,21 +28,6 @@ const (
...
-
-func GetQoSClassByAttrs(labels, annotations map[string]string) QoSClass {
-	// annotations are for old format adaption reason
-	if q, exist := labels[LabelPodQoS]; exist {
-		return GetPodQoSClassByName(q)
-	}
-	return QoSNone
-}
-
 func GetPodQoSClassByName(qos string) QoSClass {
```
Comment (jasonliu747) [inline apis/extension/qos.go:31]: 漏了一个函数？

File: apis/extension/constants.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ import corev1 "k8s.io/api/core/v1"
 
 const (
 	DomainPrefix = "koordinator.sh/"
-	// use prefix "kubernetes.io/" for extend resource
+	// ResourceDomainPrefix uses prefix "kubernetes.io/" for extend resource
```
Comment (eahydra) [inline apis/extension/constants.go:23]: 建议这里还是一起改掉吧。应该只有batch-cpu/batch-memory需要单独使用kubernetes.io前缀。不然后续会被误用。

File: apis/extension/qos.go (hunk 1 above)
Comment (saintube) [reply apis/extension/qos.go:31]: `GetPodQoSClassByName` 这个目前可以不挪，只是做类型转换的，没有其他逻辑

**Pull request merged**